### PR TITLE
Cli env var changes

### PR DESCRIPTION
**Pull request by brian-mann**
`CYPRESS_BINARY_VERSION` -> `CYPRESS_INSTALL_BINARY` (semver, url, /path/to/zip.zip)
`CYPRESS_BINARY_FOLDER` -> `CYPRESS_RUN_BINARY` (/path/to/binary)
`CYPRESS_CACHE_DIRECTORY` -> `CYPRESS_CACHE_FOLDER` (/path/to/folder)

### CODE REVIEW

File: cli/lib/errors.js [hunk 1]
```diff
@@ -94,6 +95,35 @@ const unexpected = {
...
+    solution: stripIndent`
+      You should use the equivalent environment variable CYPRESS_INSTALL_BINARY instead.
+    `,
+  },
+  CYPRESS_SKIP_BINARY_INSTALL: {
+    description: stripIndent`
+    The environment variable CYPRESS_SKIP_BINARY_INSTALL has been removed as of version ${chalk.green('3.0.0')}
+    `,
+    solution: stripIndent`
+      To skip the binary install, set CYPRESS_INSTALL_BINARY=0
```
Comment (jennifer-shehane) [inline cli/lib/errors.js]: I find this option of setting `0` to be a bit weird. Don't have an alternative suggestion, just feels weird.

File: cli/__snapshots__/install_spec.js [hunk 2]
```diff
@@ -9,7 +9,8 @@ Skipping installation:
 `
 
 exports['skip installation 1'] = `
-Skipping binary installation. Env var 'CYPRESS_SKIP_BINARY_INSTALL' was found.
+⚠ Warning: Skipping binary installation. Env var CYPRESS_INSTALL_BINARY was 0.
```
Comment (jennifer-shehane) [inline cli/__snapshots__/install_spec.js]: Sometimes you print 'env var' and other times you print 'environment variable' - I'd prefer if all errors say 'environment variable'

File: cli/lib/errors.js [hunk 3]
```diff
@@ -94,6 +95,35 @@ const unexpected = {
...
+    `,
+  },
+}
+
+const CYPRESS_RUN_BINARY = {
+  notValid: (value) => {
+    const properFormat = `**/${state.getPlatformExecutable()}`
+    return {
+      description: `Could not run binary set by env var CYPRESS_RUN_BINARY=${value}`,
```
Comment (jennifer-shehane) [inline cli/lib/errors.js]: 'env var' -> 'environment variable'

File: cli/lib/tasks/install.js [hunk 4]
```diff
@@ -144,21 +147,31 @@ const start = (options = {}) => {
...
-    const envVarVersion = process.env.CYPRESS_BINARY_VERSION
+  if (process.env.CYPRESS_INSTALL_BINARY) {
+
+    const envVarVersion = process.env.CYPRESS_INSTALL_BINARY
+    debug('using env var CYPRESS_INSTALL_BINARY %s', envVarVersion)
+
+    if (envVarVersion === '0') {
+      debug('env var CYPRESS_INSTALL_BINARY = 0, skipping install')
+      logger.log(
+        chalk.yellow(`${logSymbols.warning} Warning: Skipping binary installation. Env var CYPRESS_INSTALL_BINARY was 0.`)
```
Comment (jennifer-shehane) [inline cli/lib/tasks/install.js]: More 'env var' -> 'environment variable' ^^^

File: cli/lib/tasks/verify.js [hunk 5]
```diff
@@ -191,37 +190,69 @@ const start = (options = {}) => {
...
   })
 
-  return isMissingExecutable(binaryDir)
+  const checkEnvVar = () => {
+    debug('checking env variables')
+    if (process.env.CYPRESS_RUN_BINARY) {
+      const envBinaryPath = process.env.CYPRESS_RUN_BINARY
+      debug('CYPRESS_RUN_BINARY exists, =', envBinaryPath)
+      logger.warn(stripIndent`
+        ${logSymbols.warning} Warning: You have set the environment variable: ${chalk.white('CYPRESS_RUN_BINARY=')}${chalk.cyan(envBinaryPath)}
```
Comment (jennifer-shehane) [inline cli/lib/tasks/verify.js]: Should this say "Warning"? It's likely what they intended, yet the printout seems to indicate there's something wrong with this with the 'Warning' messaging.

File: cli/lib/tasks/verify.js [hunk 6]
```diff
@@ -191,37 +190,69 @@ const start = (options = {}) => {
...
+          This overrides the default Cypress binary version used.
+        `)
+      logger.log()
+
+      return util.isExecutableAsync(envBinaryPath)
+      .then((isExecutable) => {
+        debug('CYPRESS_RUN_BINARY is executable? :', isExecutable)
+        if (!isExecutable) {
+          return throwFormErrorText(errors.CYPRESS_RUN_BINARY.notValid(envBinaryPath))(stripIndent`
+          The supplied binary path is not executable
```
Comment (jennifer-shehane) [inline cli/lib/tasks/verify.js]: Is there any guidance we can give them here on how to move forward, like "Provide the binary in x format" or something?

File: cli/lib/errors.js (hunk 1 above)
Comment (Bkucera) [reply cli/lib/errors.js]: well, we could keep CYPRESS_SKIP_BINARY_INSTALL

File: cli/lib/tasks/verify.js (hunk 5 above)
Comment (Bkucera) [reply cli/lib/tasks/verify.js]: We similarly warn for global install, which is also probably intentional...It's just possible that a user exports the env var and has no clue what's happening...but i'd be ok with not having "Warning". Maybe change it to "Note:" ?

File: cli/lib/tasks/verify.js (hunk 6 above)
Comment (Bkucera) [reply cli/lib/tasks/verify.js]: Here's the full error printed to the user: 
image

Comment (Bkucera) [reply cli/lib/tasks/verify.js]: similarly, win32:
image

File: cli/lib/tasks/verify.js (hunk 5 above)
Comment (jennifer-shehane) [reply cli/lib/tasks/verify.js]: Yeah, I think that'd be better 👍

**Comment by jennifer-shehane** [comment]
@Bkucera A test is still failing.

**Comment by brian-mann** [comment]
@Bkucera and @jennifer-shehane are all of the docs updated using these new env vars?

**Comment by maxime1992** [comment]
:point_up_2: nop, not up to date. Currently trying to get cypress 3.x running on CI (Gitlab CI) and didn't manage to get it working after 2 hours.

Just got some help on Gitter from Gleb and he updated that repo from cypress 1 to 3 https://gitlab.com/cypress-io/cypress-example-docker-gitlab/blob/master/.gitlab-ci.yml which was really helpful. (thanks!)

I might be wrong but I think that `CYPRESS_CACHE_FOLDER` is not documented.

**Comment by Bkucera** [comment]
@maxime1992 I agree that we need to do a better job at managing documentation and version changes, as well as keeping our example repos up to date. The env var has been documented since the change was made, see here 
https://docs.cypress.io/guides/getting-started/installing-cypress.html#Overriding-the-Binary-Cache-Folder

**Pull request merged**